tox.ini

```diff
@@ -1,18 +1,20 @@
 [tox]
-minversion = 2.1
-envlist = py{27,35,36}-django{19,110,111}
+minversion = 3.1
+envlist = py{36,37,38}-django{2x,20,21}{,-faster}
 
 [travis:env]
 DJANGO =
-    1.9: django19
-    1.10: django110
-    1.11: django111
+    2.x: django2x
+    2.0: django111
+    2.1: django111
 
 [testenv]
 deps =
     -r{toxinidir}/dev-requirements.txt
-    django19: Django>=1.9,<1.10
-    django110: Django>=1.10,<1.11
-    django111: Django>=1.11,<1.12
+    django2x: Django>=2.0,<3.0
+    django20: Django>=2.0,<2.1
+    django21: Django>=2.1,<3.0
 
-commands=inv test {posargs}
+commands=
+  faster: pip install -e .[faster]
```

Comment:
This is fine, but I think the tox-y way to do it like

```
[testenv]
extras =
    faster: faster
```

Reply:
TIL, thanks!